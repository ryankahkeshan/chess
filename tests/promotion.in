setup
+ K a1 + k a8
+ R b1 + r b8
+ P b2 + p b7

+ P e7 + P f7 + P h7
+ n d8 + n g8

+ p e2 + p f2 + p h2
+ R d1 + B g1
done game human human

move h7 h8 p

move h7 h8 r

move h7 h8 b

move h7 h8 n

move h7 h8 k

move h7 h8 q

move h7 h8 P

move h7 h8 K
move h7 h8 N

move h2 h1 P

move h2 h1 R

move h2 h1 B

move h2 h1 N

move h2 h1 K

move h2 h1 Q

move h2 h1 p

move h2 h1 k
move h2 h1 b
move f7 g8 R
move f2 g1 q
move e7 d8 B
move e2 d1 n
