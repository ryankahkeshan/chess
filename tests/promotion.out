8  _ _ _ _
7 _ _ _ _ 
6  _ _ _ _
5 _ _ _ _ 
4  _ _ _ _
3 _ _ _ _ 
2  _ _ _ _
1 K _ _ _ 

  abcdefgh

8 k_ _ _ _
7 _ _ _ _ 
6  _ _ _ _
5 _ _ _ _ 
4  _ _ _ _
3 _ _ _ _ 
2  _ _ _ _
1 K _ _ _ 

  abcdefgh

8 k_ _ _ _
7 _ _ _ _ 
6  _ _ _ _
5 _ _ _ _ 
4  _ _ _ _
3 _ _ _ _ 
2  _ _ _ _
1 KR_ _ _ 

  abcdefgh

8 kr _ _ _
7 _ _ _ _ 
6  _ _ _ _
5 _ _ _ _ 
4  _ _ _ _
3 _ _ _ _ 
2  _ _ _ _
1 KR_ _ _ 

  abcdefgh

8 kr _ _ _
7 _ _ _ _ 
6  _ _ _ _
5 _ _ _ _ 
4  _ _ _ _
3 _ _ _ _ 
2  P _ _ _
1 KR_ _ _ 

  abcdefgh

8 kr _ _ _
7 _p_ _ _ 
6  _ _ _ _
5 _ _ _ _ 
4  _ _ _ _
3 _ _ _ _ 
2  P _ _ _
1 KR_ _ _ 

  abcdefgh

8 kr _ _ _
7 _p_ P _ 
6  _ _ _ _
5 _ _ _ _ 
4  _ _ _ _
3 _ _ _ _ 
2  P _ _ _
1 KR_ _ _ 

  abcdefgh

8 kr _ _ _
7 _p_ PP_ 
6  _ _ _ _
5 _ _ _ _ 
4  _ _ _ _
3 _ _ _ _ 
2  P _ _ _
1 KR_ _ _ 

  abcdefgh

8 kr _ _ _
7 _p_ PP_P
6  _ _ _ _
5 _ _ _ _ 
4  _ _ _ _
3 _ _ _ _ 
2  P _ _ _
1 KR_ _ _ 

  abcdefgh

8 kr n _ _
7 _p_ PP_P
6  _ _ _ _
5 _ _ _ _ 
4  _ _ _ _
3 _ _ _ _ 
2  P _ _ _
1 KR_ _ _ 

  abcdefgh

8 kr n _n_
7 _p_ PP_P
6  _ _ _ _
5 _ _ _ _ 
4  _ _ _ _
3 _ _ _ _ 
2  P _ _ _
1 KR_ _ _ 

  abcdefgh

8 kr n _n_
7 _p_ PP_P
6  _ _ _ _
5 _ _ _ _ 
4  _ _ _ _
3 _ _ _ _ 
2  P _p_ _
1 KR_ _ _ 

  abcdefgh

8 kr n _n_
7 _p_ PP_P
6  _ _ _ _
5 _ _ _ _ 
4  _ _ _ _
3 _ _ _ _ 
2  P _pp _
1 KR_ _ _ 

  abcdefgh

8 kr n _n_
7 _p_ PP_P
6  _ _ _ _
5 _ _ _ _ 
4  _ _ _ _
3 _ _ _ _ 
2  P _pp p
1 KR_ _ _ 

  abcdefgh

8 kr n _n_
7 _p_ PP_P
6  _ _ _ _
5 _ _ _ _ 
4  _ _ _ _
3 _ _ _ _ 
2  P _pp p
1 KR_R_ _ 

  abcdefgh

8 kr n _n_
7 _p_ PP_P
6  _ _ _ _
5 _ _ _ _ 
4  _ _ _ _
3 _ _ _ _ 
2  P _pp p
1 KR_R_ B 

  abcdefgh

8 kr n _n_
7 _p_ PP_P
6  _ _ _ _
5 _ _ _ _ 
4  _ _ _ _
3 _ _ _ _ 
2  P _pp p
1 KR_R_ B 

  abcdefgh

Invalid move
Invalid move
Invalid move
Invalid move
Invalid move
Invalid move
Invalid move
Invalid move
8 kr n _nN
7 _p_ PP_ 
6  _ _ _ _
5 _ _ _ _ 
4  _ _ _ _
3 _ _ _ _ 
2  P _pp p
1 KR_R_ B 

  abcdefgh

Invalid move
Invalid move
Invalid move
Invalid move
Invalid move
Invalid move
Invalid move
Invalid move
8 kr n _nN
7 _p_ PP_ 
6  _ _ _ _
5 _ _ _ _ 
4  _ _ _ _
3 _ _ _ _ 
2  P _pp _
1 KR_R_ Bb

  abcdefgh

8 kr n _RN
7 _p_ P _ 
6  _ _ _ _
5 _ _ _ _ 
4  _ _ _ _
3 _ _ _ _ 
2  P _pp _
1 KR_R_ Bb

  abcdefgh

8 kr n _RN
7 _p_ P _ 
6  _ _ _ _
5 _ _ _ _ 
4  _ _ _ _
3 _ _ _ _ 
2  P _p_ _
1 KR_R_ qb

  abcdefgh

8 kr B _RN
7 _p_ _ _ 
6  _ _ _ _
5 _ _ _ _ 
4  _ _ _ _
3 _ _ _ _ 
2  P _p_ _
1 KR_R_ qb

  abcdefgh

8 kr B _RN
7 _p_ _ _ 
6  _ _ _ _
5 _ _ _ _ 
4  _ _ _ _
3 _ _ _ _ 
2  P _ _ _
1 KR_n_ qb

  abcdefgh

Final Score:
White: 0
Black: 0
